### PR TITLE
Partial implementation of NoneType

### PR DESCRIPTION
**Pull request by khoobks**
This is a partial implementation of NoneType.

There are still some expected failures of tests with the pattern test_xxxx_bytes.

Current test results of the BinaryNoneTypeOperationTests is
Ran 171 tests in 143.038s
OK (expected failures=19)

The UnaryNoneTypeOperationTests should also be working.

I am not 100% sure what to do with the test names in BinaryNoneTypeOperationTest.not_implemented I have commented out the tests that cause an "unexpected success" but the remainder seem to pass. I thought I would need to remove all the test names except test_xxxx_bytes but I am clearly mistaken. So some direction here would be appreciated.

### CODE REVIEW

**Comment by khoobks** [comment]
Gagh! I'll take a look at the test failure tonight.

Closing the pull request in shame.

**Comment by freakboy3742** [comment]
@khoobks When a PR fails CI, you don't need to close the pull request - if you push an update to the same branch where the PR originated, the PR will be updated.

**Comment by cflee** [comment]
> I am not 100% sure what to do with the test names in BinaryNoneTypeOperationTest.not_implemented I have commented out the tests that cause an "unexpected success" but the remainder seem to pass. I thought I would need to remove all the test names except test_xxxx_bytes but I am clearly mistaken. So some direction here would be appreciated.

I think it is quite expected that most of the bytearray, bytes, class, complex and frozenset tests still need to be listed in not_implemented, as those types haven't really been implemented yet. Or at least that's what I did (and what was merged).

**Comment by freakboy3742** [comment]
As @cflee says - it's entirely expected that the byte array et al tests will still be in the "not_implemented" list - as soon as there's a working implementation of the byte array object itself, those tests (should) pass immediately. 

As a matter of style - delete, don't comment out the `not_implemented` entries.

File: python/common/org/python/types/NoneType.java [hunk 1]
```diff
@@ -6,6 +6,167 @@
...
+    )
+    public org.python.Object __bool__() {
+        return new org.python.types.Bool(false);
+    }
+
+    @org.python.Method(
+        __doc__ = ""
+    )
+    public org.python.Object __ge__(org.python.Object other) {
+        if(other instanceof org.python.types.Bool) {
```
Comment (freakboy3742) [inline python/common/org/python/types/NoneType.java]: Minor style point - if isn't a function call, so there should be a space after each if.

File: python/common/org/python/types/NoneType.java [hunk 2]
```diff
@@ -6,6 +6,167 @@
...
+            throw new org.python.exceptions.TypeError("unorderable types: NoneType() >= str()");
+        } else if(other instanceof org.python.types.List) {
+            throw new org.python.exceptions.TypeError("unorderable types: NoneType() >= list()");
+        } else if(other instanceof org.python.types.Set) {
+            throw new org.python.exceptions.TypeError("unorderable types: NoneType() >= set()");
+        } else if(other instanceof org.python.types.Tuple) {
+            throw new org.python.exceptions.TypeError("unorderable types: NoneType() >= tuple()");
+        } else if(other instanceof org.python.types.Dict) {
+            throw new org.python.exceptions.TypeError("unorderable types: NoneType() >= dict()");
+        }
```
Comment (freakboy3742) [inline python/common/org/python/types/NoneType.java]: Is there any way to reduce the repetition here? `Python.typeName(obj.getClass())` will give you the class name for *any* class, so you should be able to simplify quite a bit.

**Comment by khoobks** [comment]
Simplified the code as per the feedback.

**Comment by khoobks** [comment]
Erm, I don't think I had anything to do with that unexpected success

`tests.stdlib.test_sys.SysModuleTests.test_exit_from_static`

**Comment by freakboy3742** [comment]
@khoobks Yeah - not sure what is going on there. I don't get it on my local machine.

**Pull request merged**